src/main/java/com/company/linkedlist/ListNode.java

```diff
@@ -0,0 +1,29 @@
+package com.company.linkedlist;
+
+public class ListNode {
+    public int val;
+    public ListNode next;
+
+    public ListNode() {
+    }
+
+    public ListNode(int val) {
+        this.val = val;
+    }
+
+    public ListNode(int val, ListNode next) {
+        this.val = val;
+        this.next = next;
+    }
+
+    public static boolean areEqual(ListNode head1, ListNode head2) {
```

Comment:
You can move this to the test function as you are using it for comparing elements in the test

Reply:
here we are comparing two list node can we not keep in ListNode class

Reply:
Ok. Remove static in that case and keep it in the instance level. Accept only one parameter which is the head of the other list. first list will be `this`.

Reply:
getting null pointer exception

not sure how to send linked list [[]] and []

Reply:
Why one is `[[]]` and the other is `[]`? Both should be in the same depth only. If it is different, move this method some util class like `LinkedListUtils`

Reply:
Added LinkedListUtils